### PR TITLE
Code-golf rectangle_fs.wgsl to work around a suspected angle/mesa bug and avoid crash on chrome (#3931)

### PR DESCRIPTION
**Pull request by jleibs**
### What
When we added support for YUY2 format in (https://github.com/rerun-io/rerun/pull/4877) we we re-triggered the issue originally originally reported in https://github.com/rerun-io/rerun/issues/3931 and now https://github.com/rerun-io/rerun/issues/5073
 - This is a more extreme version of: https://github.com/rerun-io/rerun/pull/3948 which was the workaround the first time we hit this.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/5074/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/5074/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/5074/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/5074)
- [Docs preview](https://rerun.io/preview/e35e3a58e1f2093140d4d63479098124f6556463/docs) 
- [Examples preview](https://rerun.io/preview/e35e3a58e1f2093140d4d63479098124f6556463/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

File: crates/re_renderer/shader/rectangle_fs.wgsl
Comment (emilk) [inline crates/re_renderer/shader/rectangle_fs.wgsl:117]: This means we're always doing 4 taps, even with nearest neighbor sampling. Not great, not terrible. But worth a comment. I appreciate thought that they all point to the same tex-coord, so should be available in the texture cache.

File: crates/re_renderer/shader/decodings.wgsl
Comment (emilk) [inline crates/re_renderer/shader/decodings.wgsl:5]: it surprises me that combining these two into one function did much difference

**Review by emilk — APPROVED** [review]
So sad that this is necessary, but the resulting code is not all bad, so 👍

File: crates/re_renderer/shader/decodings.wgsl
Comment (Wumpf) [inline crates/re_renderer/shader/decodings.wgsl:11]: are you sure these are enough `WARNING!`? 😆

**Review by Wumpf — APPROVED** [review]
what Emil said. Disastrous but will do - you once again managed to not regress us, but we won't be able to hold this up for much longer.

We did not report this so far since we weren't able to break this down well enough - when building an application from javascript that uses the translated glsl shader everything works. But we have to get to the bottom of this

**Comment by emilk** [comment]
Let's make sure we test this new shader properly before merging!

File: crates/re_renderer/shader/decodings.wgsl
Comment (jleibs) [reply crates/re_renderer/shader/decodings.wgsl:5]: Yeah -- this was the final tweak that got things  to work :-(

**Comment by jleibs** [comment]
> Let's make sure we test this new shader properly before merging!
Confirmed both nv12/yuy2 working on both native and web.

**Pull request merged**
